Gradle Plugin not working if module does not have main sourceSet

**Describe the bug**
Quarkus Version: 1.7.0.Final
Quarkus Gradle fails building the application because the plugin scans the workspace and retrieves the "main" sourceSet of all modules, but there is no "main" sourceSet in some modules.

For example: I have kotlin MPP submodules where there is a javaconvention but the javaconvention has no sourceSets.

**Expected behavior**
Should not fail.

**Actual behavior**
Build fails with: 
```
Caused by: org.gradle.api.UnknownDomainObjectException: SourceSet with name 'main' not found.
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.createNotFoundException(DefaultNamedDomainObjectCollection.java:504)
        at org.gradle.api.internal.DefaultNamedDomainObjectCollection.getByName(DefaultNamedDomainObjectCollection.java:333)
        at io.quarkus.gradle.tasks.QuarkusGradleUtils.getSourceSet(QuarkusGradleUtils.java:41)
        at io.quarkus.gradle.builder.QuarkusModelBuilder.getWorkspaceModule(QuarkusModelBuilder.java:121)
        at io.quarkus.gradle.builder.QuarkusModelBuilder.getWorkspace(QuarkusModelBuilder.java:113)
        at io.quarkus.gradle.builder.QuarkusModelBuilder.buildAll(QuarkusModelBuilder.java:100)
        at io.quarkus.gradle.QuarkusPluginExtension.create(QuarkusPluginExtension.java:185)
        at io.quarkus.gradle.QuarkusPluginExtension.getQuarkusModel(QuarkusPluginExtension.java:178)
        at io.quarkus.gradle.QuarkusPluginExtension.getAppModelResolver(QuarkusPluginExtension.java:170)
        at io.quarkus.gradle.QuarkusPluginExtension.getAppModelResolver(QuarkusPluginExtension.java:166)
        at io.quarkus.gradle.QuarkusPluginExtension_Decorated.getAppModelResolver(Unknown Source)
        at io.quarkus.gradle.tasks.QuarkusBuild.buildQuarkus(QuarkusBuild.java:97)
...
```

**Suggestions for a fix:**
I have several ideas to fix this problem:
- Only scan modules that are related through dependencies to the module that is to be build.
- Ignore modules which have no "main" sourceSet
- Allow configuration of "to be excluded" sourceSets
- Allow configuration of the "main" sourceSet for each module

I think the second one is the easiest to implement, but i have no idea about the impact it would have.

--- a/devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java
+++ b/devtools/gradle/src/main/java/io/quarkus/gradle/builder/QuarkusModelBuilder.java
@@ -107,7 +107,7 @@ public class QuarkusModelBuilder implements ParameterizedToolingModelBuilder<Mod
         for (Project subproject : project.getAllprojects()) {
             final Convention convention = subproject.getConvention();
             JavaPluginConvention javaConvention = convention.findPlugin(JavaPluginConvention.class);
-            if (javaConvention == null) {
+            if (javaConvention == null || !javaConvention.getSourceSets().getNames().contains(SourceSet.MAIN_SOURCE_SET_NAME)) {
                 continue;
             }
             modules.add(getWorkspaceModule(subproject, mode));

--- a/devtools/gradle/src/test/resources/multi-module-project/nomain/build.gradle
+++ b/devtools/gradle/src/test/resources/multi-module-project/nomain/build.gradle
@@ -1,0 +1,8 @@
+plugins {
+    id 'java-library'
+}
+
+
+sourceSets {
+    remove(named("main"))
+}

--- a/devtools/gradle/src/test/resources/multi-module-project/settings.gradle
+++ b/devtools/gradle/src/test/resources/multi-module-project/settings.gradle
@@ -16,4 +16,4 @@ pluginManagement {
 }
 rootProject.name='mutli-module-project'
 
-include 'common', 'application'
+include 'common', 'application', 'nomain'

